### PR TITLE
[webview_flutter] Add zoomEnabled to webview flutter platform interface

### PR DESCRIPTION
**Pull request by NickalasB**
Since `webview_flutter` was recently updated to be a Federated Plugin, this PR is part of the [instructions on changing Federated Plugins](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changing-federated-plugins). 

This is the first-step task to add a `zoomEnabled` param on the `WebView` widget.

If this is approved, merged, and published, it will be fast-followed by two other small PRs updating `webview_flutter_wkwebview` and `webview_flutter_android`.

I have been maintaining https://github.com/flutter/plugins/pull/3325 for a decent chunk of time and I'm hoping following these new Federated plugin guidelines can get this puppy over the finish line and we can get off our Forked version of this plugin.

*List which issues are fixed by this PR. You must list at least one issue.*
Implements flutter/flutter#70731
Implements flutter/flutter#48283
Relates to flutter/flutter#48245
Relates to flutter/flutter#60921

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*
- [x] No this is *not* a breaking change

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.0.1
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md:1]: 1.1.0

File: packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ repository: https://github.com/flutter/plugins/tree/master/packages/webview_flut
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+webview_flutter%22
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 1.0.0
+version: 1.0.1
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml:7]: (And here obviously.)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Just a version nit, otherwise looks good.

**Comment by stuartmorgan** [comment]
@bparrishMines @mvanbeusekom Could one of you take a look as well, per the newly-documented [two-committer policy](https://github.com/flutter/flutter/wiki/Tree-hygiene#getting-a-code-review)?

File: packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (NickalasB) [reply packages/webview_flutter/webview_flutter_platform_interface/CHANGELOG.md]: Fixed in dc60b83

File: packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml (hunk 2 above)
Comment (NickalasB) [reply packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml]: Fixed in dc60b83

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by NickalasB** [comment]
> LGTM

Thank you again @stuartmorgan. Is there a way to force a new CI build? The failure is a timeout and seems unrelated to my changes

**Comment by stuartmorgan** [comment]
I've restarted the failing test. (There was an issue with that test across the repo at the end of last week.)

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
This should automatically be published within a couple of hours, unblocking the next PR (or PRs; you can do the two implementation changes as either one PR or two as you prefer, since they don't depend on each other). Thanks again!

**Pull request merged**